File: dagger2/src/main/java/pgreze/dagger2/analytics/AnalyticsManager.java
package pgreze.dagger2.analytics;

import javax.inject.Inject;
import javax.inject.Singleton;

@Singleton
public class AnalyticsManager {

    private final MessageService messageService;

    /**
     * Constructor with @Inject will be used by dagger
     * @param messageService
     */
    @Inject
    public AnalyticsManager(MessageService messageService) {
        this.messageService = messageService;
    }

    public void event(String event) {
        messageService.showMessage("Register " + event);
    }
}

File: dagger/android/src/main/kotlin/pgreze/dagger2/analytics/AnalyticsManager.kt
package pgreze.dagger2.analytics

import javax.inject.Inject
import javax.inject.Singleton

/**
 * Constructor with @Inject will be used by dagger.
 * @param messageService a singleton scoped only dependency
 */
@Singleton
class AnalyticsManager @Inject constructor(
        private val messageService: MessageService
) {
    fun event(event: String) {
        messageService.showMessage("Register " + event)
    }
}
